File: apps/web/src/types/ticket-activity.ts
export interface TicketTimestamps {
  created_at: string;
  updated_at: string | null;
  gdpr_consent_at: string | null;
  assigned_at: string | null;
  in_progress_at: string | null;
  on_hold_at: string | null;
  first_response_at: string | null;
  resolved_at: string | null;
  closed_at: string | null;
  cancelled_at: string | null;
  last_escalation_at: string | null;
  response_due_at: string | null;
  resolution_due_at: string | null;
}

export interface TicketActivityItem {
  id: string;
  occurred_at: string;
  event_type: string;
  label_da: string;
  actor_display_name: string | null;
  visibility: "internal" | "external" | "system";
  detail: string | null;
}

File: apps/api/src/star_itsm_api/schemas/ticket_activity.py
from datetime import datetime
from typing import Literal
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class TicketTimestampsRead(BaseModel):
    """Milestone timestamps for a ticket lifecycle."""

    model_config = ConfigDict(from_attributes=True)

    created_at: datetime
    updated_at: datetime | None = None
    gdpr_consent_at: datetime | None = None
    assigned_at: datetime | None = None
    in_progress_at: datetime | None = None
    on_hold_at: datetime | None = None
    first_response_at: datetime | None = None
    resolved_at: datetime | None = None
    closed_at: datetime | None = None
    cancelled_at: datetime | None = None
    last_escalation_at: datetime | None = None
    response_due_at: datetime | None = None
    resolution_due_at: datetime | None = None


class TicketActivityItemRead(BaseModel):
    id: UUID
    occurred_at: datetime
    event_type: str
    label_da: str
    actor_display_name: str | None = None
    visibility: Literal["internal", "external", "system"] = "external"
    detail: str | None = None
